### PR TITLE
Update the reference app with a new Amplify + React UI, one click deployment of frontend

### PR DESCRIPTION
**Pull request by angelarw**
*Issue #, if available:*

*Description of changes:* 
* Built a new Amplify + React frontend with full user management
* One click deploy of both frontend and backend using Amplify console and CloudFormation
* Updated README and arch diagrams
* Added AppSync as the central API coordination 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: amplify/backend/api/photoshare/parameters.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "AppSyncApiName": "photoshare",
+    "DynamoDBBillingMode": "PAY_PER_REQUEST",
+    "DynamoDBEnableServerSideEncryption": "false",
+    "AuthCognitoUserPoolId": {
+        "Fn::GetAtt": [
+            "authphotoshareb143529b",
```
[the remark this answers is not in the document]
Comment (rts-rob) [reply amplify/backend/api/photoshare/parameters.json:7]: Generated by Amplify

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+STATE_MACHINE_JSON = state-machine.json
+TEMPLATE_NAME = image-processing
+
+
+init: ##=> Install OS deps and dev tools
+	$(info [*] Initializing...)
+	@$(MAKE) _install_os_packages
+
+
+deploy: ##=> Deploy services
```
[the remark this answers is not in the document]
Comment (angelarw) [reply Makefile:17]: updated. thanks for the suggestion!

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,50 @@
+AWS_BRANCH ?= dev
+STACK_NAME ?= photo-sharing
+GRAPHQL_API_ID ?= "UNDEFINED"
+PHOTO_BUCKET ?= "UNDEFINED"
+STATE_MACHINE_JSON = state-machine.json
```
[the remark this answers is not in the document]
Comment (angelarw) [reply Makefile]: updated. thanks for the suggestion!

File: README.md [hunk 4]
```diff
@@ -7,174 +7,74 @@ This repository contains sample code for all the Lambda functions depicted in th
...
-cd ../cloudformation
 
-python inject_state_machine_cfn.py -s state-machine.json -c image-processing.serverless.yaml -o image-processing.complete.yaml
+### How to deploy
+Follow these instructions to deploy the application (both backend and frontend):
 
-aws cloudformation package --region $REGION --s3-bucket $S3BUCKET --template image-processing.complete.yaml --output-template-file image-processing.output.yaml
-```
-Last, deploy the stack with the resulting yaml (`image-processing.output.yaml `) through the CloudFormation Console or command line:
+[![One-click deployment](https://oneclick.amplifyapp.com/button.svg)](https://console.aws.amazon.com/amplify/home#/deploy?repo=https://github.com/angelarw/lambda-refarch-imagerecognition)
```
[the remark this answers is not in the document]
Comment (angelarw) [reply README.md:24]: :)

File: README.md [hunk 5]
```diff
@@ -7,174 +7,74 @@ This repository contains sample code for all the Lambda functions depicted in th
...
 
-```bash
-aws cloudformation deploy --region $REGION --template-file image-processing.output.yaml --stack-name photo-sharing-backend --capabilities CAPABILITY_IAM
-```
+1. Use **1-click deployment** button above
+1. If you don't have an IAM Service Role, create one
```
[the remark this answers is not in the document]
Comment (angelarw) [reply README.md]: added explanation and screenshot

File: Workshop/step-1.md [hunk 6]
```diff
@@ -80,8 +80,13 @@ When you are constructing a state machine definition by authoring JSON directly,
 
 	</details>
 
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (angelarw) [reply Workshop/step-1.md]: cleaned up. thanks!

File: amplify.yml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        - npm run build
+  artifacts:
+    baseDirectory: src/react-frontend/build
+    files:
+      - "**/*"
+  cache:
+    paths:
+      - node_modules/**/*
+      - src/react-frontend/node_modules/**/*
+#  customHeaders:
```
[the remark this answers is not in the document]
Comment (angelarw) [reply amplify.yml]: deleted

File: amplify/team-provider-info.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
+{
```
[the remark this answers is not in the document]
Comment (angelarw) [reply amplify/team-provider-info.json:1]: cleaned it to {}

File: amplify/backend/function/S3Trigger984fb593/src/.env [hunk 9]
```diff
@@ -0,0 +1,3 @@
+API_PHOTOSHARE_GRAPHQLAPIENDPOINTOUTPUT=https://6fn4ef454jgnniaxiv6ojtk3ty.appsync-api.us-east-1.amazonaws.com/graphql
```
[the remark this answers is not in the document]
Comment (angelarw) [reply amplify/backend/function/S3Trigger984fb593/src/.env]: removed

**Pull request merged**